### PR TITLE
Check modules and prospectors settings when reload is off

### PR DESCRIPTION
**Pull request by exekias**
This change ensures that we check configs when loading them through
`filebeat.config.modules`, `filebeat.config.prospectors` or `metricbeat.config.modules`.

It will error and exit if settings are wrong and **reload is disabled**

Part of #4810

### CODE REVIEW

**Comment by kvch** [comment]
Please note that it conflicts with my PR #5026, so one of us will have to rebase their branch after merging. :)

File: libbeat/cfgfile/reload.go [hunk 1]
```diff
@@ -63,13 +66,60 @@ func NewReloader(cfg *common.Config) *Reloader {
...
 	return &Reloader{
 		registry: NewRegistry(),
 		config:   config,
+		path:     path,
 		done:     make(chan struct{}),
 	}
 }
 
+// Check configs are valid (only if reload is disabled)
+func (rl *Reloader) Check(runnerFactory RunnerFactory) error {
```
Comment (kvch) [inline libbeat/cfgfile/reload.go]: It is quite similar to the existing config reload code in `Run` function. I would extract anything that could be extracted into smaller functions. So those could be reused it in both `Check` and `Run`. Thus, code duplication could be eliminated.

**Comment by kvch** [comment]
I have one concerns regarding the code. I have already accepted your PR, so decide if you would address it or not.

File: libbeat/cfgfile/reload.go (hunk 1 above)
Comment (exekias) [reply libbeat/cfgfile/reload.go]: I'll do some refactoring, thanks!

**Comment by exekias** [comment]
This should be ready for a second look

**Comment by kvch** [comment]
After it turns green, I would merge it.

**Pull request merged**